bundles/ubuntu22.04/bundle.sh.gotmpl

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -euo pipefail
+export DEBIAN_FRONTEND=noninteractive
+apt-get update && apt-get install -y --no-install-recommends \
+apt-transport-https software-properties-common dpkg-dev curl gpg
+
+KUBERNETES_DEB_BUILD_VERSION="1.1"
+CRI_TOOLS_DEB_BUILD_VERSION="1.1"
+
+# Install dependencies.
+mkdir -p /etc/apt/keyrings
+curl -fsSL "https://pkgs.k8s.io/core:/stable:/{{ .KubernetesMajorMinorVersion }}/deb/Release.key" | gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
+echo "deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/{{ .KubernetesMajorMinorVersion }}/deb/ /" | tee /etc/apt/sources.list.d/kubernetes.list
+apt-get update
+#shellcheck disable=SC2046
+apt-cache depends --recurse --no-recommends --no-suggests --no-conflicts --no-breaks --no-replaces --no-enhances --no-pre-depends $(< packages.txt) | grep "^\w" > /tmp/packages
+
+# The list returned by apt-cache depends does not contain the versions
+# Append the bare Kubernetes packages with the provided versions
+sed -i 's/kubelet/kubelet='"{{ .KubernetesVersion }}-${KUBERNETES_DEB_BUILD_VERSION}"'/' /tmp/packages
+sed -i 's/kubectl/kubectl='"{{ .KubernetesVersion }}-${KUBERNETES_DEB_BUILD_VERSION}"'/' /tmp/packages
+sed -i 's/kubeadm/kubeadm='"{{ .KubernetesVersion }}-${KUBERNETES_DEB_BUILD_VERSION}"'/' /tmp/packages
+sed -i 's/cri-tools/cri-tools='"{{ .CRIToolsVersion }}-${CRI_TOOLS_DEB_BUILD_VERSION}"'/' /tmp/packages
+# https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4745 pins version for CAPA
+# enable it back if https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4745, https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/5115 are resolved
+echo 'cloud-init=23.1.2-0ubuntu0~22.04.1' >> /tmp/packages
```

Comment:
This will add two cloud-init packages in the bundle. one latest and one with this specific version. 
Since the package-bundle command is provider agnostic, I think it should be okay to have to version of a packages in bundle instead of adding provider specific logic. 

If you are fine with this, I will make the same changes for `ununtu-20.04` too in separate PR.

Reply:
i think we can handle this another way and should probably move away from this sort of pinning altogether

Reply:
please rebase with the new changes that removes needs for the pin